src/components/tools/Settings/UploadBoundary.svelte

```diff
@@ -8,103 +8,159 @@
+    } else {
+      handleValidityError("Must be a GeoJSON FeatureCollection or Feature.");
+      return null;
     }
+  }
+
+  async function processUpload(file) {
+    handleLoading();
 
-    if (geojson && geojsonError && geojson.features.length === 1) {
-      fileUploadProps.errorSubject = "Warning";
-      fileUploadProps.errorBody =
-        "We can fetch data for this area but cannot display it on the inset map.";
-      const location = formatFeature(geojson.features[0], "custom");
-      location.title = file.name;
-      dispatch("upload", { location });
+    // NOTE: this only validates the file against the Cal-Adapt API.
+    // The response is not used for anything.
+    const [, validityError] = await handleXHR(validateShape(file));
+
+    if (validityError) {
+      handleValidityError(validityError);
       return;
     }
 
-    if (geojson && geojson.features.length > 1) {
-      fileUploadProps.status = "edit";
-      fileUploadProps.invalid = true;
-      fileUploadProps.errorSubject = "Error";
-      fileUploadProps.errorBody =
-        "The uploaded file has more than 1 geometry. The tool can display fetch data for only 1 geometry at a time.";
-      return;
+    // Try converting the uploaded boundary to JSON format
+    const [geojson, geojsonError] = await handle(convertFileToGeojson(file));
+    const isValidGeoJson = typeof geojson === "object" && "type" in geojson;
+
+    if (geojsonError) {
+      if (isValidGeoJson) {
+        handleValidityError(
+          `We can fetch data for this area, but the boundary may not display on
+          the location map.`,
+          "Warning"
+        );
+        // NOTE: we don't early return here so that the custom boundary may still be used
+      } else {
+        handleValidityError(geojsonError);
+        return;
+      }
+    }
+
+    if (isValidGeoJson) {
+      const feature = handleGeoJsonSuccess(geojson);
+      if (feature) {
+        const location = formatFeature(feature, "custom");
+        location.title = file.name;
+        dispatch("upload", { location });
+      }
     }
   }
 </script>
 
 <style>
-  :global(.bx--file__selected-file) {
+  div :global(.bx--file__selected-file) {
     background: #f1f4f8;
   }
-  :global(.bx--file__selected-file .bx--file-filename) {
+
+  div :global(.bx--file__selected-file .bx--file-filename) {
     margin-bottom: 0;
   }
 </style>
 
-<FileUploader
-  {...fileUploadProps}
-  kind="tertiary"
-  bind:this="{fileUploader}"
-  on:add="{({ detail }) => {
-    processUpload(detail[0]);
-  }}"
-  on:click="{clearFiles}"
-/>
-{#if fileUploadProps.errorSubject === "Warning" || fileUploadProps.errorSubject === "Error"}
-  <span>{fileUploadProps.errorBody}</span>
-{/if}
+<div>
+  <FileUploader
+    {...fileUploadProps}
+    kind="tertiary"
+    bind:this="{fileUploader}"
+    bind:files
+    on:add="{({ detail }) => {
+      processUpload(detail[0]);
+    }}"
+    on:click="{clearFiles}"
+  />
+
+  {#if fileUploadProps.errorSubject}
```

Comment:
maybe this should be `{#if fileUploadProps.errorSubject || fileUploadProps.errorBody}`?